include/mfmg/dealii/amge_host.templates.hpp

```diff
@@ -185,6 +186,34 @@ AMGe_host<dim, MeshEvaluator, VectorType>::compute_local_eigenvectors(
     solver.solve(agglomerate_system_matrix, agglomerate_mass_matrix,
                  inv_system_matrix, eigenvalues, eigenvectors);
   }
+  else if (_eigensolver_type == "lanczos")
+  {
+    boost::property_tree::ptree lanczos_params;
+#if 0
+    lanczos_params.put("is_deflated", true);
+    lanczos_params.put("num_eigenpairs_per_cycle", 2);
+    lanczos_params.put("num_cycles", 3);
+#else
+    lanczos_params.put("is_deflated", false);
+    lanczos_params.put("num_eigenpairs", n_eigenvectors);
+#endif
+    lanczos_params.put("max_iterations", 200);
+    lanczos_params.put("tolerance", tolerance);
+    lanczos_params.put("percent_overshoot", 5);
+
+    Lanczos<dealii::SparseMatrix<double>, dealii::Vector<double>> solver(
+        agglomerate_system_matrix, lanczos_params);
+
+    solver.solve();
+    ASSERT(n_eigenvectors == static_cast<unsigned int>(solver.num_evecs()),
+           "Wrong number of computed eigenpairs");
+
+    // Copy the eigenvalues and the eigenvectors
+    for (unsigned int i = 0; i < n_eigenvectors; ++i)
+      eigenvalues[i] = solver.get_eval(i);
```

Comment:
Replace loop by `eigenvalues = solver.get_evals();`

Reply:
No, the have different type (`complex` vs `double`).